core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -1606,6 +1574,25 @@ class DAGSchedulerSuite
     assertDataStructuresEmpty()
   }
 
+  test("accumulators are updated on exception failures") {
```

Comment:
Is this a change from existing behavior?

(This is a reminder to myself to come back later in the review and look more carefully)